build.sh

```diff
@@ -41,11 +41,13 @@ do
       # run lang-specific tests
       (cd lang/java; ./build.sh test)
       # install java artifacts required by other builds and interop tests
-      mvn install -DskipTests
+      mvn -B install -DskipTests
       (cd lang/py; ./build.sh test)
       (cd lang/py3; ./build.sh test)
       (cd lang/c; ./build.sh test)
-      (cd lang/c++; ./build.sh test)
+      # The C++ tests are failing
+      # https://issues.apache.org/jira/projects/AVRO/issues/AVRO-2230
```

Comment:
AVRO-2230 is now marked as fixed in Jira.

Reply:
Still failing :'(
```
Scanning dependencies of target avrocpp_s
[  1%] Building CXX object CMakeFiles/avrocpp_s.dir/impl/Compiler.cc.o
/testptch/unknown/lang/c++/impl/Compiler.cc:174:15: warning: â€˜std::string avro::nameof(const NodePtr&)â€™ defined but not used [-Wunused-function]
 static string nameof(const NodePtr& n)
               ^
[  2%] Building CXX object CMakeFiles/avrocpp_s.dir/impl/Node.cc.o
[  4%] Building CXX object CMakeFiles/avrocpp_s.dir/impl/NodeImpl.cc.o
/testptch/unknown/lang/c++/impl/NodeImpl.cc: In function â€˜std::string avro::{anonymous}::escape(const string&)â€™:
/testptch/unknown/lang/c++/impl/NodeImpl.cc:31:8: warning: â€˜autoâ€™ changes meaning in C++11; please remove it [-Wc++0x-compat]
   for (auto c : unescaped) {
        ^
/testptch/unknown/lang/c++/impl/NodeImpl.cc:31:13: error: â€˜câ€™ does not name a type
   for (auto c : unescaped) {
             ^
/testptch/unknown/lang/c++/impl/NodeImpl.cc:67:3: error: expected â€˜;â€™ before â€˜returnâ€™
   return s;
   ^
/testptch/unknown/lang/c++/impl/NodeImpl.cc:67:3: error: expected primary-expression before â€˜returnâ€™
/testptch/unknown/lang/c++/impl/NodeImpl.cc:67:3: error: expected â€˜;â€™ before â€˜returnâ€™
/testptch/unknown/lang/c++/impl/NodeImpl.cc:67:3: error: expected primary-expression before â€˜returnâ€™
/testptch/unknown/lang/c++/impl/NodeImpl.cc:67:3: error: expected â€˜)â€™ before â€˜returnâ€™
In file included from /usr/include/boost/assert.hpp:51:0,
                 from /usr/include/boost/format/parsing.hpp:21,
                 from /usr/include/boost/format.hpp:50,
                 from /testptch/unknown/lang/c++/api/Exception.hh:24,
                 from /testptch/unknown/lang/c++/api/Node.hh:28,
                 from /testptch/unknown/lang/c++/api/GenericDatum.hh:29,
                 from /testptch/unknown/lang/c++/api/NodeImpl.hh:23,
                 from /testptch/unknown/lang/c++/impl/NodeImpl.cc:21:
/testptch/unknown/lang/c++/impl/NodeImpl.cc: In member function â€˜virtual void avro::NodeRecord::printJson(std::ostream&, int) constâ€™:
/testptch/unknown/lang/c++/impl/NodeImpl.cc:247:59: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     assert(defaultValues.empty() || (defaultValues.size() == fields));
                                                           ^
/testptch/unknown/lang/c++/impl/NodeImpl.cc: In member function â€˜virtual void avro::NodePrimitive::printDefaultToJson(const avro::GenericDatum&, std::ostream&, int) constâ€™:
/testptch/unknown/lang/c++/impl/NodeImpl.cc:292:13: error: â€˜to_stringâ€™ is not a member of â€˜stdâ€™
       os << std::to_string(g.value<float>());
             ^
/testptch/unknown/lang/c++/impl/NodeImpl.cc:295:13: error: â€˜to_stringâ€™ is not a member of â€˜stdâ€™
       os << std::to_string(g.value<double>());
             ^
/testptch/unknown/lang/c++/impl/NodeImpl.cc: In function â€˜std::string avro::{anonymous}::escape(const string&)â€™:
/testptch/unknown/lang/c++/impl/NodeImpl.cc:68:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
CMakeFiles/avrocpp_s.dir/build.make:100: recipe for target 'CMakeFiles/avrocpp_s.dir/impl/NodeImpl.cc.o' failed
make[2]: *** [CMakeFiles/avrocpp_s.dir/impl/NodeImpl.cc.o] Error 1
make[1]: *** [CMakeFiles/avrocpp_s.dir/all] Error 2
CMakeFiles/Makefile2:425: recipe for target 'CMakeFiles/avrocpp_s.dir/all' failed
make: *** [all] Error 2
Makefile:147: recipe for target 'all' failed
```

Reply:
I've created [a new ticket](https://issues.apache.org/jira/browse/AVRO-2249) on Jira, the tests are still failing :(